api/app/src/main/kotlin/packit/security/provider/TokenProvider.kt

```diff
@@ -0,0 +1,47 @@
+package packit.security.provider
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Component
+import packit.AppConfig
+import packit.security.profile.UserPrincipal
+import java.time.Duration
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+
+interface JwtIssuer
+{
+    fun issue(authentication: Authentication): String
+}
+
+@Component
+class TokenProvider(val config: AppConfig) : JwtIssuer
+{
+    companion object
+    {
+        const val TOKEN_ISSUER = "packit-api"
+        const val TOKEN_AUDIENCE = "packit"
+    }
+
+    override fun issue(authentication: Authentication): String
+    {
+        val user = authentication.principal as UserPrincipal
+
+        val roles = user.authorities.map { it.authority }
+
+        val createdDate = Instant.now()
+
+        val expiredDate = createdDate.plus(Duration.of(config.authExpiryDays, ChronoUnit.DAYS))
```

Comment:
We can add refresh token implementation to the solution,  users will have to re-login when token expires. Credentials do not expire because it's for username and password => basic user login method.